frontend-react/src/components/Admin/AdminReceiverDashboard.tsx

```diff
@@ -457,56 +500,165 @@ function MainRender(props: {
         } // for dayslots
 
         // <editor-fold desc="Per-Receiver render">
-        const showRow =
-            props.filterRowStatus !== SuccessRate.UNDEFINED
-                ? props.filterRowStatus === successForRow.currentState
-                : true;
-
-        if (showRow) {
-            // we saved the start of this block of data, grab the information from there.
-            const key = keys[keyOffsetStartRow];
-            const orgName = props.data[key].organizationName;
-            const recvrName = props.data[key].receiverName;
-            const successRate = Math.round(
-                (100 * successForRow.countSuccess) /
-                    (successForRow.countSuccess + successForRow.countFailed)
-            );
-            const titleClassName =
-                SUCCESS_RATE_CLASSNAME_MAP[successForRow.currentState];
-
-            perReceiverElements.push(
-                <Grid
-                    row
-                    key={`perreceiver-row-${keyOffsetStartRow}`}
-                    className={"perreceiver-row"}
-                >
-                    <Grid className={`title-column ${titleClassName}`}>
-                        <div className={"title-text"}>
-                            {orgName}
-                            <br />
-                            {recvrName}
-                            <br />
-                            {successRate}%
-                        </div>
-                    </Grid>
-                    <ScrollSyncPane enabled>
-                        <Grid row className={"horizontal-scroll"}>
-                            <Grid row className={"week-column"}>
-                                {perDayElements}
-                            </Grid>
-                        </Grid>
-                    </ScrollSyncPane>
+        // we saved the start of this block of data, grab the information from there.
+        const orgName = props.data[offsetStartRow].organizationName;
+        const recvrName = props.data[offsetStartRow].receiverName;
+        const combinedName = `${orgName} ${recvrName}`.toLowerCase();
+        const successRate = Math.round(
+            (100 * successForRow.countSuccess) /
+                (successForRow.countSuccess + successForRow.countFailed)
+        );
+        const titleClassName =
+            SUCCESS_RATE_CLASSNAME_MAP[successForRow.currentState];
+
+        // if we have error text, then check if we filtered out all the slices
+        const allSlicesFiltered = filterErrorText.length
+            ? visibleSliceCount === 0
+            : false;
+
+        // cheat confession: using `data-` props allow us to stick properties
+        // on components without type checking requiring it be a formal prop.
```

Comment:
yeah but there are some nice things about using props. Why not make this a component? I have a sneaking suspicion that when things get split out into actual components you see better re-render performance, but that could just be superstition

Reply:
I didn't make them slice component because of fatigue. I also figured by sticking them in data html props, react wouldn't feel the need to get involved with them and it would improve performance.

But now I'm curious.